scripts/generate_hamilton_input_quant_studio.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+
+import csv
+
+from EPPs.common import InvalidStepError, StepEPP
+
+
+class GenerateHamiltonInputQuantStudio(StepEPP):
+    # define the rows and columns in the input plate (standard 96 well plate pattern)
+    plate_rows = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H']
+    plate_columns = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']
+    csv_column_headers = ['Input Plate', 'Input Well', 'Output Plate', 'Output Well', 'DNA Volume', 'TE Volume']
+    _max_nb_input_containers = 27
+    _max_nb_output_containers = 1
+
+    def __init__(self, argv=None):
+        """ additional argument required for the location of the Hamilton input file so def __init__ customised."""
+        super().__init__(argv)
+        self.hamilton_input = self.cmd_args.hamilton_input
+
+        assert self.csv_column_headers is not None, 'csv_column_headers needs to be set by the child class'
+        assert self._max_nb_input_containers is not None, 'number of permitted input containers needs to be set ' \
+                                                          'by the child class'
+        assert self._max_nb_output_containers is not None, 'number of permitted output containers needs to be set ' \
+                                                           'by the child class'
+
+    @staticmethod
+    def add_args(argparser):
+
+        argparser.add_argument(
+            '-i', '--hamilton_input', nargs='*', help='Add up to 3 file spaces in LIMS step for Hamilton inputs')
```

Comment:
`hamilton_input` should be pluralised here and in the variable name to make it clear it is a list.
I think you'd  need to have at least one file so change `nargs` to `nargs='+'`

Reply:
updated